src/circuit.rs

```diff
@@ -306,7 +308,7 @@ where
 
         // Add ProverKey to Prover
         prover.prover_key = Some(prover_key.clone());
-        prover.prove(&ck)
+        prover.prove(&ck, &mut rng)
```

Comment:
We take `rng` as `mut rng: &mut R`. Mutable references are copy, even if the referred value isn't copy (they are just pointers).
```suggestion
        prover.prove(&ck, rng)
```

The current code works because `&mut R` is virtually congruent to `&mut &mut R`, but I believe this is a nit. We shouldn't take the reference as mutable, since we don't change the reference itself, but the data it points to (RNG).